src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/CachedModelMetadata.cs

```diff
@@ -82,6 +83,17 @@ protected CachedModelMetadata(CachedModelMetadata<TPrototypeCache> prototype, Fu
             PrototypeCache = prototypeCache;
         }
 
+        // Sealing for consistency with other properties.
+        // ModelMetadata already caches the collection and we have no use case for a ComputeAdditionalValues() method.
+        /// <inheritdoc />
+        public sealed override IDictionary<string, object> AdditionalValues
```

Comment:
we could pull forward infrastructure similar to MVC 5's `IMetadataAware` and `AdditionalMetadataAttribute`.   but it's not clear that work is required.  (it's not that horrendous to create you're own `DataAnnotationsModelMetadataProvider` or `AssociatedMetadataProvider` subclass.)  further `IMetadataAware` doesn't support the user's scenario in #1758 because the `IMetadataAware` implementation doesn't get the applicable attributes.

all up, moving more forward from MVC 5 is beyond the scope of #1758 and this PR.  I could file a follow-up "needs design" enhancement if you feel strongly about this.